CMakeLists.txt

```diff
@@ -13,7 +13,35 @@ if(NOT Poco_FOUND)
   if(DEFINED CMAKE_BUILD_TYPE)
     list(APPEND extra_cmake_args -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE})
   endif()
-  if(WIN32)
+  if(DEFINED CMAKE_TOOLCHAIN_FILE)
+    list(APPEND extra_cmake_args "-DCMAKE_TOOLCHAIN_FILE=${CMAKE_TOOLCHAIN_FILE}")
+    if(DEFINED ANDROID)
+      if(DEFINED ANDROID_ABI)
+        list(APPEND extra_cmake_args "-DANDROID_ABI=${ANDROID_ABI}")
+      endif()
+      if(DEFINED ANDROID_CPP_FEATURES)
+        list(APPEND extra_cmake_args "-DANDROID_CPP_FEATURES=${ANDROID_CPP_FEATURES}")
+      endif()
+      if(DEFINED ANDROID_FUNCTION_LEVEL_LINKING)
+        list(APPEND extra_cmake_args "-DANDROID_FUNCTION_LEVEL_LINKING=${ANDROID_FUNCTION_LEVEL_LINKING}")
+      endif()
+      if(DEFINED ANDROID_NATIVE_API_LEVEL)
+        list(APPEND extra_cmake_args "-DANDROID_NATIVE_API_LEVEL=${ANDROID_NATIVE_API_LEVEL}")
+      endif()
+      if(DEFINED ANDROID_NDK)
+        list(APPEND extra_cmake_args "-DANDROID_NDK=${ANDROID_NDK}")
+      endif()
+      if(DEFINED ANDROID_STL)
+        list(APPEND extra_cmake_args "-DANDROID_STL=${ANDROID_STL}")
+      endif()
+      if(DEFINED ANDROID_TOOLCHAIN_NAME)
+        list(APPEND extra_cmake_args "-DANDROID_TOOLCHAIN_NAME=${ANDROID_TOOLCHAIN_NAME}")
+      endif()
+      list(APPEND extra_cmake_args
+        -DPOCO_UNBUNDLED:BOOL=OFF
+        -DCMAKE_C_FLAGS="-Wno-shift-negative-value")
+    endif()
+  elseif(WIN32)
```

Comment:
Wouldn't that clutter `extra_cmake_args` a bit? If I understood you correctly, you suggest something like this:

```cmake
if(WIN32)
  ...
  list(APPEND extra_cmake_args -DPOCO_UNBUNDLED:BOOL=OFF)
else()
  ...
  list(APPEND extra_cmake_args -DPOCO_UNBUNDLED:BOOL=ON)
endif()

if(DEFINED CMAKE_TOOLCHAIN_FILE)
  ...
  list(APPEND extra_cmake_args -DPOCO_UNBUNDLED:BOOL=OFF)
endif()
```
so, on Android, `extra_cmake_args` would end up containing both `-DPOCO_UNBUNDLED:BOOL=OFF` and `-DPOCO_UNBUNDLED:BOOL=ON` and that to me looks a bit fragile, since if any of the `if` clauses are moved, that would have a side-effect on the order in which CMake evaluates `-DPOCO_UNBUNDLED:BOOL`